tests/integration/test_security.py

```diff
@@ -178,7 +178,7 @@ def test_cors_apigw_not_applied(self, aws_client):
                 "Origin": "https://app.localstack.cloud",
             },
         )
-        assert response.status_code == 404
+        assert response.status_code in (403, 404)
```

Comment:
nit: Maybe we could make this parameterized to already take the proper return code in case of the new provider (same below)?

Reply:
Thing is, I'm not too sure yet what is the appropriate status code, I believe it is 403. 

How would we parametrize this?
Shouldn't we instead use the config variable to do the assert? `if config.APIGW_NEXT_GEN_PROVIDER: assert 403 else assert 404`?

Reply:
Well, if we don't know it currently then no.
I had the impression that the old code (404) is not in parity with AWS and the new provider will (403).
If this is the case, then we should make sure that the assert for the old provider is properly removed.
However, if we don't know this yet, then it's fine for now. ;)